From 6ad0474adc7be9dd32e7eee6f6511b830cf24e6f Mon Sep 17 00:00:00 2001
From: Subhash Jadavani <subhashj@codeaurora.org>
Date: Wed, 5 Jul 2017 17:52:13 -0700
Subject: [PATCH 17/40] libdiskconfig: handle UFS card

SD card have native sector size of 512 bytes where as UFS card have the
native sector size of 4096 bytes. Let's allow sector size of 4096 bytes
as well in order to support UFS cards.
---
 libdiskconfig/diskconfig.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/libdiskconfig/diskconfig.c b/libdiskconfig/diskconfig.c
index c7e1b4374..236f79b08 100644
--- a/libdiskconfig/diskconfig.c
+++ b/libdiskconfig/diskconfig.c
@@ -302,11 +302,17 @@ validate(struct disk_info *dinfo)
             goto fail;
         }
 
-        if (!sect_sz || sect_sz != dinfo->sect_size) {
-            ALOGE("Device sector size is zero or sector sizes do not match!");
+        if (!sect_sz) {
+            ALOGE("Device sector size is zero!");
             goto fail;
         }
 
+        if ((sect_sz != 512) && (sect_sz != 4096)) {
+            ALOGE("Device sector size %d is not supported", sect_sz);
+            goto fail;
+        }
+        dinfo->sect_size = sect_sz;
+
         /* allow the user override the "disk size" if they provided num_lba */
         if (!dinfo->num_lba) {
             if (ioctl(fd, BLKGETSIZE64, &disk_size) < 0) {
-- 
2.11.0

